core/tx_pool.go

```diff
@@ -55,6 +56,15 @@ var (
 	evictionInterval     = time.Minute   // Time interval to check for evictable transactions
 )
 
+var (
+	// Metrics
+	pendingDiscardMeter    = metrics.NewMeter("core/txpool/transactions/pending/discard")
+	pendingReplaceMeter    = metrics.NewMeter("core/txpool/transactions/pending/replace")
+	queuedDiscardMeter     = metrics.NewMeter("core/txpool/transactions/queued/discard")
+	queuedReplaceMeter     = metrics.NewMeter("core/txpool/transactions/queued/replace")
+	invalidTxMeter         = metrics.NewMeter("core/txpool/transactions/invalid")
```

Comment:
I would also add a couple more metrics:

 * `txpool/pending/ratelimit`: when there are too many transactions in total and we start dropping them
 * `txpool/queued/ratelimit`: when there are too many transactions in total and we start dropping them
 * `txpool/pending/nofunds`: when already pending transaction are dropped due to funds running out
 * `txpool/queued/nofunds`: when already queued transaction are dropped due to funds running out

These could be valuable to see if someone's deliberately trying to exhaust resources (`ratelimit`) or tries to queue up txs that become invalid later (`nofunds`)